src/storage-controller/src/lib.rs

```diff
@@ -935,6 +935,81 @@ where
         Ok(())
     }
 
+    async fn alter_ingestion_export_data_configs(
+        &mut self,
+        // Primary ingestion id, source-export id, and new data config
+        source_exports: Vec<(GlobalId, GlobalId, SourceExportDataConfig)>,
```

Comment:
I have one final nit which is that having the caller supply the `ingestion_id` is redundant, since we can directly look up the export and get the ingestion id from its `DataSource` value. Then you can turn this back into a `BTreeMap<GlobalId, SourceExportDataConfig>` which just lists exports